Karma-Khameleons/bangazon-website
# Sam customer order

## Description
This pull request ties in all of the functionality of viewing order details and closing an order. Closed orders have their "active_order" property switched from 1 to 0. Currently, the order detail view will also create a new order for the user if they have no active order



## Related Ticket(s)
Partially resolves #8 
Fixed: #9 


## Problem to Solve
Users should be able to see an overview of their current order and checkout an order.

## Changes
Added content to admin.py to allow myself to manually add items to orders


## Expected Behavior
When a user who has items in their current order goes to the order view, they should be able to see a list of said items, their prices, the order total, a list of payment type options, and an affordance to complete an order.
If the user has no items in their order, or no payment types created, they are prompted to go shop for items or create a new payment type.


## Steps to Test Solution
1. From the root of the django project, run the following commands:
```
git fetch --all
git checkout origin/sam-CustomerOrder
rm -rf bang_app/migrations
python manage.py makemigrations
python manage.py migrate
python manage.py test bang_app -v 2
python manage.py runserver
```
1. From the bangazon website, go to ```/admin``` to add products to a customer's order.
1. As that customer, navigate to ```/order``` and complete the ordering process
**(Note, if you get an error saying something like Django found 2 instances of customerOrder, that will be because you manually created a second active order for that customer. To fix this, simply change one of the customerOrder's "active_order" value to 0 from the ```/admin``` page. This bug should only present itself when people manually create a new order from the admin page for a user that already has an active order.)**
1. Confirm that you are redirected to the order success view


## Unit Testing
- [] I certify that all existing tests pass 
- [X] I certify that all existing tests pass which are tied to this code
## Documentation


## Deploy Notes

1. Migrations will have to be remade, as detailed in the Steps to Test section.
1. At one point in development I had to delete and recreate my database. I don't THINK that should be necessary at this point, but if you're having problems testing it then that is probably where we should start looking for the problem.
